### ISSUE TITLE
New Rule: no-implicit-type-coercion

### ISSUE THREAD

**Issue by adidahiya** [palantir/tslint#625]
I'd like to forbid null checks which use implicit type coercion:

``` ts
if (nonBooleanValue) {
    ...
}
```

and instead enforce checks like this:

``` ts
if (nonBooleanValue != null) {
    ...
}
```

**Comment by adidahiya** [comment]
blocked on complete type information, which will be enabled by #680

**Comment by adidahiya** [comment]
related: #253

**Comment by andy-hanson** [comment]
`strict-boolean-expressions` should do this.

**Comment by OliverJAsh** [comment]
`strict-boolean-expressions` does slightly more than just disallowing boolean coercion. I filed a suggestion for a new rule which only disallows boolean coercion: https://github.com/palantir/tslint/issues/4495.
